src/aspire/estimation/covar2d.py

```diff
@@ -9,25 +9,107 @@
 
 from aspire.utils.matrix import shrink_covar
 from aspire.utils.optimize import conj_grad
-from aspire.denoise.covar2d import RotCov2D
-from aspire.utils import ensure
 
+from aspire.utils import ensure
 
 logger = logging.getLogger(__name__)
 
 
-class Cov2DCTF(RotCov2D):
+class RotCov2D:
     """
-    Define a derived class for denoising 2D images using CTF information and the Covariance Wiener Filtering (CWF)
-    Cov2D method described in
+    Define a class for denoising 2D images using the Cov2D method with CTF information described in
 
     T. Bhamre, T. Zhang, and A. Singer, "Denoising and covariance estimation of single particle cryo-EM images",
     J. Struct. Biol. 195, 27-81 (2016). DOI: 10.1016/j.jsb.2016.04.013
     """
 
+    def __init__(self, basis, as_type='single'):
+        """
+        constructor of an object for 2D covariance analysis
+        """
+        self.basis = basis
+        ensure(basis.d == 2, 'Only two-dimensional basis functions are needed.')
+        self.as_type = as_type
+
+    def get_mean(self, coeffs):
+        """
+        Calculate the mean vector from the expansion coefficients of 2D images without CTF information.
+
+        :param coeffs: A coefficient vector (or an array of coefficient vectors) to be averaged.
+        :return: The mean value vector for all images.
+        """
+        if coeffs is None:
```

Comment:
Why would a user send in `None` as the coefficients?

Reply:
Initially this is to make sure that a non-empty `coeffs` array is passed. Now change to check the size.

Reply:
I don't think that's necessary. If you try to call a function without the required number of arguments, it already gives an error. Why would someone send in an array of size zero?